tools/odbc/include/odbc_fetch.hpp

```diff
@@ -2,33 +2,83 @@
 #define ODBC_FETCH_HPP
 
 #include "duckdb.hpp"
-// #include <cstdint>
 #include <sqltypes.h>
+#include <sql.h>
+#include <sqlext.h>
+#include <vector>
+#include <stack>
 
 namespace duckdb {
 
 struct OdbcHandleStmt;
 
-enum class FetchOrientation : uint8_t { COLUMN = 0, ROW = 1 };
+enum class FetchBindingOrientation : uint8_t { COLUMN = 0, ROW = 1 };
 
 class OdbcFetch {
 public:
-	FetchOrientation orientation;
-	SQLULEN rows_to_fetch;
+	FetchBindingOrientation bind_orientation;
+	SQLULEN rowset_size;
 	SQLPOINTER row_length;
 	SQLUSMALLINT *row_status_buff;
 
+	const static SQLULEN SINGLE_VALUE_FETCH = 1;
+
+	SQLULEN cursor_type;
+	SQLLEN row_count;
+
+private:
+	// main structure to hold the fetched chunks
+	std::vector<unique_ptr<DataChunk>> chunks;
+	// stack chunk's pointers to be used during fetch prior
+	std::stack<DataChunk *> stack_prior_chunks;
```

Comment:
Is a stack necessary here? Can't we have an index to the current entry of the chunks array? (e.g. `idx_t current_chunk_idx`)?

Reply:
Done.